File: src/main/java/cz/solutions/cockroach/SaleRecord.java
package cz.solutions.cockroach;

import lombok.Value;
import org.joda.time.LocalDate;

import java.util.Optional;

@Value
public class SaleRecord {
    LocalDate date;
    String type;
    double quantity;
    double salePrice;
    double purchasePrice;
    double purchaseFmv;
    LocalDate purchaseDate;
    Optional<String> grantId;

    public boolean isTaxable(){
        return date.isBefore(purchaseDate.plusYears(3));
    }
}

File: src/main/kotlin/cz/solutions/cockroach/SaleRecord.kt
package cz.solutions.cockroach

import org.joda.time.LocalDate

data class SaleRecord(
    val date: LocalDate,
    val type: String,
    val quantity: Double,
    val salePrice: Double,
    val purchasePrice: Double,
    val purchaseFmv: Double,
    val purchaseDate: LocalDate,
    val grantId: String?,
    val symbol: String,
    val broker: String,
) {
    fun isTaxable(): Boolean {
        return date.isBefore(purchaseDate.plusYears(3))
    }
}